ocs_ci/ocs/machine.py

```diff
@@ -138,6 +140,113 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
     return False
 
 
+def create_custom_machineset(
+    role='app', instance_type='m4.xlarge', label='app-scale', zone='a'
+):
+    """
+    Function to create custom machineset works only for AWS
+    i.e. Using this user can create nodes with different instance type and role.
+    https://docs.openshift.com/container-platform/4.1/machine_management/creating-machineset.html
+
+    Args:
+        role (str): Role type to be added for node eg: it will be app,worker
+        instance_type (str): Type of aws instance
+        label (str): Label to be added to the node
+        zone (str): Machineset zone for node creation.
+
+    Raise:
+        UnsupportedPlatformError: Incase of wrong platform
+    """
+    # check for platform, since it's supported only for IPI
+    if config.ENV_DATA['deployment_type'] == 'ipi':
+        machinesets_obj = OCP(kind=constants.MACHINESETS, namespace=constants.OPENSHIFT_MACHINE_API_NAMESPACE)
+        for machine in machinesets_obj.get()['items']:
+            # Get inputs from existing machineset config.
+            region = machine.get('spec').get('template').get('spec').get('providerSpec').get('value').get(
+                'placement').get('region')
+            aws_zone = machine.get('spec').get('template').get('spec').get('providerSpec').get('value').get(
+                'placement').get('availabilityZone')
+            cls_id = machine.get('spec').get('selector').get('matchLabels').get(
+                'machine.openshift.io/cluster-api-cluster')
+            ami_id = machine.get('spec').get('template').get('spec').get('providerSpec').get('value').get(
+                'ami').get('id')
+            if aws_zone == f"{region}{zone}":
+                machineset_yaml = templating.load_yaml(
```

Comment:
Can not find the 'else' for that if.

Reply:
This `if` itself inside for loop for each existing machineset data. 
i.e. Below default machineset of OCS cluster.
```
rperiyas-cls-w76c9-worker-us-east-2a 1 1 1 1 144m
rperiyas-cls-w76c9-worker-us-east-2b 1 1 1 1 144m
rperiyas-cls-w76c9-worker-us-east-2c 1 1 1 1 144m
```
Custom machineset.yaml required params are gathered from particular zone matching existing machineset. The 'else' would be just for not matching zones which is not required.

Why we need particular machineset ?
The reason would be to fetch aws zone, region, ami image id and cluster name.